pkg/utils/k8sutil.go

```diff
@@ -210,6 +210,9 @@ func EnsureK8sResources(ns, name string, funcObj *spec.Function, client kubernet
 	labels := map[string]string{
 		"function": name,
 	}
+	for k, v := range funcObj.Metadata.Labels {
```

Comment:
yes this is the concern. If we think it should be allowed then it still work. I have also updated the labels to deploy, svc and cm then it should not a problem.